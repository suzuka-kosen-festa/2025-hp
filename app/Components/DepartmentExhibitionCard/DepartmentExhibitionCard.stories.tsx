import type { Meta, StoryObj } from "@storybook/react";
import DepartmentExhibitionCard from "./DepartmentExhibitionCard";

type T = typeof DepartmentExhibitionCard;

const meta: Meta<T> = {
  title: "DepartmentExhibitionCard",
  component: DepartmentExhibitionCard,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  argTypes: {
    department: {
      control: "text",
      description: "学科名",
    },
    title: {
      control: "text",
      description: "タイトル",
    },
    description: {
      control: "text",
      description: "展示の説明文",
    },
  },
} satisfies Meta<T>;

type Story = StoryObj<T>;

export const Default: Story = {
  args: {
    department: "学科名",
    title: "タイトル",
    description: "説明文",
  },
};

export default meta;
