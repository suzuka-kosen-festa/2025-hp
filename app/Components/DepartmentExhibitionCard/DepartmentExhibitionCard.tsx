import Box from "@mui/material/Box";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import Chip from "@mui/material/Chip";
import Typography from "@mui/material/Typography";

export interface DepartmentExhibitionCardProps {
  department: string
  title: string
  description: string
}

// 学科別テーマカラーの定義
const departmentColors: Record<string, string> = {
  機械工学科: "#E783AC",
  電気電子工学科: "#AE904A",
  電子情報工学科: "#89D0DD",
  生物応用科学科: "#6E9D45",
  材料工学科: "#002670",
};

export default function DepartmentExhibitionCard(props: DepartmentExhibitionCardProps) {
  const { department, title, description } = props;
  const themeColor = departmentColors[department] || "#4a90e2"; // デフォルトカラー
  return (
    <Card sx={{
      width: "100%",
      maxWidth: "100%",
      height: "100%",
      display: "flex",
      flexDirection: "column",
      border: `2px solid ${themeColor}`,
      boxShadow: `0 0 15px ${themeColor}40`,
      borderRadius: { xs: "12px", sm: "16px" },
    }}
    >
      <Box
        sx={{
          height: "fit-content",
          display: "flex",
          flexDirection: "column",
          alignItems: "stretch",
          justifyContent: "flex-start",
          flexGrow: 1,
        }}
      >
        <CardContent sx={{ p: { xs: 1.5, sm: 2 }, display: "flex", flexDirection: "column", flexGrow: 1 }}>
          <Chip
            label={department}
            size="small"
            sx={{
              "mb": 1,
              "backgroundColor": themeColor,
              "color": "white",
              "fontWeight": "bold",
              "fontSize": { xs: "0.7rem", sm: "0.75rem" },
              "& .MuiChip-label": {
                textShadow: "0 0 5px rgba(255, 255, 255, 0.3)",
              },
            }}
          />
          <Typography
            variant="h6"
            fontWeight="bold"
            gutterBottom
            sx={{
              fontSize: { xs: "1.1rem", sm: "1.25rem" },
            }}
          >
            {title}
          </Typography>
          <Typography
            variant="body2"
            color="text.secondary"
            sx={{
              fontSize: { xs: "0.8rem", sm: "0.875rem" },
              lineHeight: { xs: 1.4, sm: 1.6 },
            }}
          >
            {description}
          </Typography>
        </CardContent>
      </Box>
    </Card>
  );
}
