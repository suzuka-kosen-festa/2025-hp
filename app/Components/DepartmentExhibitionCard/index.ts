export { default } from "./DepartmentExhibitionCard";
export type { DepartmentExhibitionCardProps } from "./DepartmentExhibitionCard";
