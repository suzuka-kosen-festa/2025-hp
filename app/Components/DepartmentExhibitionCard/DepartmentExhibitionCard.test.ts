// @vitest-environment jsdom
import { composeStories } from "@storybook/react";
import { expect, it } from "vitest";
import * as stories from "./DepartmentExhibitionCard.stories";

const { Default } = composeStories(stories);

it("departmentExhibitionCard snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
