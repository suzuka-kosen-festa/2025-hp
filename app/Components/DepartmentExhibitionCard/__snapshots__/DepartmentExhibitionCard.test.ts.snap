// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`departmentExhibitionCard snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-dj2uoc-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiBox-root css-19idkm8"
    >
      <div
        class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-rj74di-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
          >
            学科名
          </span>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1jrq8e9-MuiTypography-root"
        >
          タイトル
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1wle3ir-MuiTypography-root"
        >
          説明文
        </p>
      </div>
    </div>
  </div>
</div>
`;
